app/src/pages/playground/PlaygroundChatTemplate.tsx

```diff
@@ -41,22 +61,31 @@ export function PlaygroundChatTemplate(props: PlaygroundChatTemplateProps) {
     throw new Error(`Invalid template type ${template.__type}`);
   }
 
+  const sensors = useSensors(
+    useSensor(PointerSensor),
+    useSensor(KeyboardSensor, {
+      coordinateGetter: sortableKeyboardCoordinates,
+    })
+  );
+
   return (
-    <DragDropProvider
-      onDragOver={(event) => {
-        const newMessages = move(template.messages, event);
-        updateInstance({
-          instanceId: id,
-          patch: {
-            template: {
-              __type: "chat",
-              messages: newMessages,
-            },
-          },
-        });
-      }}
-      onDragEnd={(event) => {
-        const newMessages = move(template.messages, event);
+    <DndContext
+      sensors={sensors}
+      onDragEnd={({ active, over }) => {
+        if (!over || active.id === over.id) {
+          return;
+        }
```

Comment:
can you help me understand this?

Reply:
if the dragged element not over any other element, or it's over itself, no need to reorder the list so we cut out early, pulled this from an example, probably works without it, just an optimization i suppose

Reply:
Gotcha